sdk/src/main/java/co/decodable/sdk/pipeline/DecodableStreamSourceBuilder.java

```diff
@@ -8,26 +8,31 @@
 package co.decodable.sdk.pipeline;
 
 import co.decodable.sdk.pipeline.util.Incubating;
+import org.apache.flink.api.common.serialization.DeserializationSchema;
 
 /** Builder for creating {@literal DecodableStreamSource} instances. */
 @Incubating
-public interface DecodableStreamSourceBuilder {
+public interface DecodableStreamSourceBuilder<T> {
 
   /**
    * Specifies the name of the stream to read from. Either this or {@link #withStreamId(String)} may
    * be used, but not both.
    */
-  DecodableStreamSourceBuilder withStreamName(String streamName);
+  DecodableStreamSourceBuilder<T> withStreamName(String streamName);
 
   /**
    * Specifies the id of the stream to read from. Either this or {@link #withStreamName(String)} may
    * be used, but not both.
    */
-  DecodableStreamSourceBuilder withStreamId(String streamId);
+  DecodableStreamSourceBuilder<T> withStreamId(String streamId);
 
   /** Specifies the start-up mode to use when reading from the stream. */
-  DecodableStreamSourceBuilder withStartupMode(StartupMode startupMode);
+  DecodableStreamSourceBuilder<T> withStartupMode(StartupMode startupMode);
+
+  /** Specifies the deserialization schema to be used. */
+  DecodableStreamSourceBuilder<T> withDeserializationSchema(
```

Comment:
I would distinguish between key and value deserialization schema

Reply:
Interesting question, do we really want to do this? I don't think we allow any direct access to the key via SQL pipelines (only by designating specific fields of the value as key column(s)), so I'm having some doubts whether we should open this up via custom pipelines. I'd prefer to keep this question for another time.

Reply:
Okay, my concern was around forward compatibility -- If we introduce key / value deserialization schemas, then we have to deprecate the `deserializationSchema` methods.

But I think that's fine.

Reply:
We're going to do a Beta release, plus all the types are annotated with `@Incubating`, so we'll still be at liberty to change this, even without prior deprecation.